src/xmlwriter.rs

```diff
@@ -63,15 +62,27 @@ impl XMLWriter {
     }
 
     // Write an XML start tag with attributes.
-    pub(crate) fn xml_start_tag_attr(&mut self, tag: &str, attributes: &Vec<(&str, String)>) {
-        let mut attribute_str = "".to_string();
+    pub(crate) fn xml_start_tag_attr(&mut self, tag: &str, attributes: &[(&str, String)]) {
+        let approx_len = tag.len() + 6 * attributes.len() + 2;
```

Comment:
This could probably be a fixed value of 64 or  128 bytes. I could instruments the code and work out the most common length.

Reply:
Those `.len()` calls are O(1) because the length is included in the type. Also, addition and subtraction operations are extremely fast. In other words, that line probably takes less than 5 nanoseconds.

Reply:
It would be better to get the average len of any single attribute pair. Right now I just conservatively assume its 2. That's where the 6 comes from, 2 + 4 = 6 for each attribute.